packages/rich-text/src/useOnValueChanged.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback, useMemo } from 'react';
+
+import { toContentfulDocument } from '@contentful/contentful-slatejs-adapter';
+import { Document } from '@contentful/rich-text-types';
+import { PlateEditor } from '@udecode/plate-core';
+import debounce from 'lodash/debounce';
+import { Operation } from 'slate';
+
+import schema from './constants/Schema';
+
+/**
+ * Returns whether a given operation is relevant enough to trigger a save.
+ */
+const isRelevantOperation = (op: Operation) => {
+  if (op.type === 'set_selection') {
+    return false;
+  }
+
+  return true;
+};
+
+export type OnValueChangedProps = {
+  editor: PlateEditor;
+  handler?: (value: Document) => unknown;
+};
+
+export const useOnValueChanged = ({ editor, handler }: OnValueChangedProps) => {
+  const onChange = useMemo(
+    () =>
+      debounce((document: unknown) => {
+        handler?.(toContentfulDocument({ document, schema }));
+      }, 500),
```

Comment:
I'd prefer this delay to be configurable, to easily refine it later and/or tweak it in different apps. In future we might even think about making it dynamic to some extent, for example basing it on the content size or on the history of the user typing happening at the same time of the handler run

Reply:
I think it's unlikely that we will make use of the configuration given that it has always been that way for years. The way I thought of this is basically: we can always make it configurable later easily but not the other way around (not without a major version bump)

> tweak it in different apps

Do you have a case in mind?

Reply:
Not any specific one, but, for example, I imagine that compose is more likely to face performance issue than the web app. Thus a more relaxed debounce there would be beneficial.
As you said, though, it's easy to implement later